ambry-api/src/main/java/com.github.ambry/clustermap/StateTransitionException.java

```diff
@@ -42,6 +42,10 @@ public TransitionErrorCode getErrorCode() {
     /**
      * If bootstrap process fails at some point for specific replica.
      */
-    BootstrapFailure
+    BootstrapFailure,
```

Comment:
Could you add javadoc for the enum class that explains how TransitionErrorCode is used by Helix or elsewhere?  I only see them used by the tests.

Reply:
Sure, for now it is used by tests to determine which step has failed. In the future, it will probably be used by Helix State Model when investigating specific replica that is in `ERROR` state.